tests/unit/ReportDataCollectorTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * @author Sujith Haridasan <sharidasan@owncloud.com>
+ *
+ * @copyright Copyright (c) 2018, ownCloud GmbH
+ * @license AGPL
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License, version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License, version 2,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\ConfigReport\Tests;
+
+use OC\IntegrityCheck\Checker;
+use OC\SystemConfig;
+use OC\User\Manager;
+use OCA\ConfigReport\ReportDataCollector;
+use OCP\IAppConfig;
+use OCP\IGroupManager;
+use Test\TestCase;
+
+/**
+ * Class ReportDataCollectorTest
+ *
+ * @group DB
+ * @package OCA\ConfigReport\Tests
+ */
+class ReportDataCollectorTest extends TestCase {
+	private $integrityChecker;
+	private $userManager;
+	private $groupManager;
+	private $sysConfig;
+	private $appConfig;
+	private $connection;
+	private $reportDataCollector;
+
+	protected function setUp() {
+		parent::setUp();
+
+		$this->integrityChecker = $this->createMock(Checker::class);
+		$this->userManager = $this->createMock(Manager::class);
+		$this->groupManager = $this->createMock(IGroupManager::class);
+		$this->sysConfig = $this->createMock(SystemConfig::class);
+		$this->appConfig = $this->createMock(IAppConfig::class);
+		//$this->connection = $this->createMock(IDBConnection::class);
+		$this->connection = \OC::$server->getDatabaseConnection();
+
+		$this->reportDataCollector = new ReportDataCollector($this->integrityChecker,
+				$this->userManager, $this->groupManager, [], "1.0",
+				'1', 'foo', $this->sysConfig, $this->appConfig,
+				$this->connection);
+	}
+
+	public function testGetOcMigrationArray() {
+		$results = $this->invokePrivate($this->reportDataCollector, 'getOcMigrationArray', []);
+		foreach ($results as $result) {
+			$this->assertArrayHasKey('app', $result);
+			$this->assertArrayHasKey('version', $result);
+			$values = \array_values($result);
+			$this->assertContains($values[0], ['core', 'dav', 'files_sharing', 'files_trashbin', 'notifications', 'dav', 'files_external', 'federatedfilesharing', 'customgroups']);
```

Comment:
this will likely fail as we don't bundle "customgroups", etc

only pick the ones present in the core repository

Reply:
I have updated the test, removed `customgroups`. I can't delete `notifications`, though. As the bundle `owncloud-daily-master-qa.tar.bz2` has notifications app.